packages/financial-templates-lib/src/logger/DiscordTransport.ts

```diff
@@ -59,18 +76,58 @@ export class DiscordTransport extends Transport {
 
       // Send webhook request to each of the configured webhooks upstream. This posts the messages on Discord. Execute
       // these sequentially with a soft delay of 2 seconds between calls to avoid hitting the discord rate limit.
-      if (webHooks.length)
-        for (const webHook of webHooks) {
-          await axios.post(webHook, body);
-          await delay(2);
-        }
+      if (webHooks.length) for (const webHook of webHooks) this.logQueue.push({ webHook, body });
+
+      await this.executeLogQueue(); // Start processing the log que.
     } catch (error) {
       console.error("Discord error", error);
     }
 
     callback();
   }
 
+  // Processes a queue of logs produced by the transport. Executes sequentially and listens to the response from the
+  // Discord API to back off and sleep if we are exceeding their rate limiting. Sets the parent transports isFlushed
+  // variable to block the bot from closing until the whole queue has been flushed.
+  async executeLogQueue(): Promise<void> {
+    if (this.isQueueBeingExecuted) return; // If the queue is currently being executed, return.
+    this.isQueueBeingExecuted = true; // Set the queue to being executed.
+    // Set the parent isFlushed to false to prevent the logger from closing while the queue is being executed. Note this
+    // is separate variable from the isQueueBeingExecuted flag as this isFlushed is global for the logger instance.
+    (this as any).parent.isFlushed = false;
+
+    // If the previous iteration set a backOffDuration then wait for this duration.
+    if (this.backOffDuration != 0) await delay(this.backOffDuration);
```

Comment:
nit: why do we store backOffDuration on the class? Why not just store it as a local variable declared at the beginning of this method? Is it relevant to anything outside this run?